lib/galaxy/jobs/runners/util/cli/job/slurm.py

```diff
@@ -61,20 +87,79 @@ def parse_single_status(self, status, job_id):
             # Job still on cluster and has state.
             id, state = status[1].split()
             return self._get_job_state(state)
+        elif len(status) <= 1:
+            log.debug(f"For job '{job_id}', relying on 'sacct' method to determine job state")
+            # Job no longer on cluster, retrieve state
+            pdata = subprocess.run(
```

Comment:
All of the cli runners use abstractions to run commands via different plugins (various flavors of shells and ssh), that's why things like `get_single_status` return strings. See for instance https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/cli.py#L176. If you use subprocess here you're breaking that abstraction. I would perhaps change the signature of all parse_single_status functions to allow passing in the shell.

Reply:
it looks like there's more than one command you want to run ? I'd keep it simple and pass in the shell interface.